src/mobile/src/main/java/tupperdate/android/ui/home/profile/ProfileGeneral.kt

```diff
@@ -0,0 +1,181 @@
+package tupperdate.android.ui.home.profile
+
+import androidx.compose.foundation.ScrollableRow
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import dev.chrisbanes.accompanist.coil.CoilImage
+import tupperdate.android.R
+import tupperdate.android.data.legacy.api.RecipeApi
+import tupperdate.android.ui.theme.TupperdateTheme
+import tupperdate.android.ui.theme.components.ProfilePicture
+import tupperdate.android.ui.theme.material.BrandedButton
+import tupperdate.android.ui.theme.modifier.shade
+import java.util.List.of
+
+@Composable
+fun ProfileGeneral(
+    name: String,
+    email: String,
+    image: Any,
+    location: String,
+    userRecipes: List<RecipeApi.Recipe>,
+    onEditClick: () -> Unit,
+    onLocationChange: (String) -> Unit,
+    onNewRecipeClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    //TODO add the new appbar done by G-L for the new navigation
+    Column(modifier = modifier.fillMaxSize().padding(16.dp)) {
+        Text(
+            text = stringResource(id = R.string.profile_title_capital),
+            style = MaterialTheme.typography.overline
+        )
+        ProfileRecap(name = name, email = email, image = image, onEditClick = onEditClick)
+        OutlinedTextField(
+            value = location,
+            onValueChange = onLocationChange,
+            modifier = Modifier.fillMaxWidth()
+                .padding(top = 4.dp, bottom = 24.dp),
+            label = { Text(stringResource(R.string.profile_location)) },
+            placeholder = { location }
+        )
+        Text(
+            text = stringResource(id = R.string.profile_tupps),
+            style = MaterialTheme.typography.overline
+        )
+        Row(
+            modifier = Modifier.fillMaxWidth(1f)
+                .padding(top = 16.dp),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            BrandedButton(
+                value = stringResource(id = R.string.profile_new_recipe), onClick = onNewRecipeClick,
+                modifier = Modifier.width((recipeCardWidth * 0.8).dp)
```

Comment:
Eventually `fillMaxWidth(0.8)` (the variant which takes a percentage as a parameter). But I'm not sure you'd really want that, you probably should use a padding instead.